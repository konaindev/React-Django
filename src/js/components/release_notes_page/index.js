import React, { Component } from "react";
import PropTypes from "prop-types";

import Container from "../container";
import PageChrome from "../page_chrome";
import ReleaseNotesTable from "../release_notes_table";
import SectionHeader from "../section_header";

export const ReleaseNotesPage = ({ release_notes }) => (
  <PageChrome>
    <Container>
      <SectionHeader title="Releases" smallMarginTop />
      <ReleaseNotesTable release_notes={release_notes} />
    </Container>
  </PageChrome>
);

ReleaseNotesTable.propTypes = {
  release_notes: PropTypes.array.isRequired
};

export default ReleaseNotesPage;
