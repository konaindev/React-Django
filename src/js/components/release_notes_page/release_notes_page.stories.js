import React from "react";
import { storiesOf } from "@storybook/react";

import ReleaseNotesPage from "./index";

const props = {
  release_notes: [
    {
      id: 1,
      title: "Alderaan",
      version: "2.13.18",
      date: "2018-11-11",
      content: "content"
    },
    {
      id: 2,
      title: "Alderaan",
      version: "2.14.12",
      date: "2018-12-03",
      content: "content"
    },
    {
      id: 3,
      title: "Alderaan",
      version: "2.15.1",
      date: "2018-12-11",
      content: "content"
    },
    {
      id: 4,
      title: "Alderaan",
      version: "2.16.3",
      date: "2018-12-29",
      content: "content"
    },
    {
      id: 5,
      title: "Alderaan",
      version: "2.16.18",
      date: "2019-01-11",
      content: "content"
    },
    {
      id: 6,
      title: "Alderaan",
      version: "2.17.2",
      date: "2019-02-03",
      content: "content"
    },
    {
      id: 7,
      title: "Alderaan",
      version: "2.18.1",
      date: "2019-02-28",
      content: "content"
    }
  ]
};

storiesOf("ReleaseNotesPage", module).add("default", () => (
  <ReleaseNotesPage {...props} />
));
