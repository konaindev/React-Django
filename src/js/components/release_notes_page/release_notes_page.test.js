import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import ReleaseNotesPage from "./index";
import { props } from "./release_notes_page.stories";

describe("ReleaseNotesPage", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ReleaseNotesPage {...props} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
