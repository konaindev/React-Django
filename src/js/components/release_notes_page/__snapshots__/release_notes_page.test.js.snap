// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseNotesPage renders correctly 1`] = `
<div
  className="chrome"
>
  <div
    className="page-chrome__top-chrome"
  >
    <div
      className="page-chrome__top-chrome__content"
    >
      <div
        className="page-header"
      >
        <div
          className="container page-header__inner"
        >
          <div
            className="remarkably-logo"
          >
             
          </div>
        </div>
      </div>
    </div>
    <div
      className="page-chrome__top-chrome__spacing-fix"
      style={
        Object {
          "height": 76,
        }
      }
    >
       
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="section-header section-header--mt-sm"
    >
      <p
        className="section-header__title"
      >
        Releases
      </p>
    </div>
    <div
      className="panel panel--default"
    >
      <table
        className="release-notes-table"
      >
        <tr
          className="release-notes-table__row"
        >
          <td
            className="release-notes-table__col release-notes-table__version"
          >
            2.13.18
          </td>
          <td
            className="release-notes-table__col release-notes-table__title"
          >
            Alderaan
          </td>
          <td
            className="release-notes-table__col release-notes-table__date"
          >
            11/11/18
          </td>
          <td
            className="release-notes-table__col"
          >
            <a
              className="release-notes-table__link"
              href="/releases/1"
            >
              NOTES 
              <big>
                ›
              </big>
            </a>
          </td>
        </tr>
        <tr
          className="release-notes-table__row"
        >
          <td
            className="release-notes-table__col release-notes-table__version"
          >
            2.14.12
          </td>
          <td
            className="release-notes-table__col release-notes-table__title"
          >
            Alderaan
          </td>
          <td
            className="release-notes-table__col release-notes-table__date"
          >
            12/3/18
          </td>
          <td
            className="release-notes-table__col"
          >
            <a
              className="release-notes-table__link"
              href="/releases/2"
            >
              NOTES 
              <big>
                ›
              </big>
            </a>
          </td>
        </tr>
        <tr
          className="release-notes-table__row"
        >
          <td
            className="release-notes-table__col release-notes-table__version"
          >
            2.15.1
          </td>
          <td
            className="release-notes-table__col release-notes-table__title"
          >
            Alderaan
          </td>
          <td
            className="release-notes-table__col release-notes-table__date"
          >
            12/11/18
          </td>
          <td
            className="release-notes-table__col"
          >
            <a
              className="release-notes-table__link"
              href="/releases/3"
            >
              NOTES 
              <big>
                ›
              </big>
            </a>
          </td>
        </tr>
        <tr
          className="release-notes-table__row"
        >
          <td
            className="release-notes-table__col release-notes-table__version"
          >
            2.16.3
          </td>
          <td
            className="release-notes-table__col release-notes-table__title"
          >
            Alderaan
          </td>
          <td
            className="release-notes-table__col release-notes-table__date"
          >
            12/29/18
          </td>
          <td
            className="release-notes-table__col"
          >
            <a
              className="release-notes-table__link"
              href="/releases/4"
            >
              NOTES 
              <big>
                ›
              </big>
            </a>
          </td>
        </tr>
        <tr
          className="release-notes-table__row"
        >
          <td
            className="release-notes-table__col release-notes-table__version"
          >
            2.16.18
          </td>
          <td
            className="release-notes-table__col release-notes-table__title"
          >
            Alderaan
          </td>
          <td
            className="release-notes-table__col release-notes-table__date"
          >
            1/11/19
          </td>
          <td
            className="release-notes-table__col"
          >
            <a
              className="release-notes-table__link"
              href="/releases/5"
            >
              NOTES 
              <big>
                ›
              </big>
            </a>
          </td>
        </tr>
        <tr
          className="release-notes-table__row"
        >
          <td
            className="release-notes-table__col release-notes-table__version"
          >
            2.17.2
          </td>
          <td
            className="release-notes-table__col release-notes-table__title"
          >
            Alderaan
          </td>
          <td
            className="release-notes-table__col release-notes-table__date"
          >
            2/3/19
          </td>
          <td
            className="release-notes-table__col"
          >
            <a
              className="release-notes-table__link"
              href="/releases/6"
            >
              NOTES 
              <big>
                ›
              </big>
            </a>
          </td>
        </tr>
        <tr
          className="release-notes-table__row"
        >
          <td
            className="release-notes-table__col release-notes-table__version"
          >
            2.18.1
          </td>
          <td
            className="release-notes-table__col release-notes-table__title"
          >
            Alderaan
          </td>
          <td
            className="release-notes-table__col release-notes-table__date"
          >
            2/28/19
          </td>
          <td
            className="release-notes-table__col"
          >
            <a
              className="release-notes-table__link"
              href="/releases/7"
            >
              NOTES 
              <big>
                ›
              </big>
            </a>
          </td>
        </tr>
      </table>
    </div>
  </div>
</div>
`;
